src/main/scala/org/broadinstitute/dsde/rawls/jobexec/MethodConfigResolver.scala

```diff
@@ -0,0 +1,95 @@
+
+object MethodConfigResolver {
+
+  private def getSingleResult[T](seq: Seq[T]): Try[T] = {
+    seq.headOption match {
+      case None => Failure(new EmptyResultException)
+      case Some(null) => Failure(new EmptyResultException) // sometimes expression eval returns a Seq(null) for an empty result
+      case Some(notNull) => Success(notNull)
+    }
+  }
+
+  /**
+   * Evaluate expressions in a method config against a root entity.
+   * @return map from input name to a Try containing a resolved value or failure
+   */
+  def resolveInputs(methodConfig: MethodConfiguration, entity: Entity, wdl: String, txn: RawlsTransaction): Map[String, Try[Any]] = {
+    txn withGraph { graph =>
+      val evaluator = new ExpressionEvaluator(graph, new ExpressionParser)
+      methodConfig.inputs.map {
+        case (name, expression) => {
+          val evaluated = evaluator.evaluate(entity.workspaceName.namespace, entity.workspaceName.name, entity.entityType, entity.name, expression)
+          // now look at expected WDL type to see if we should unpack a single value from the Seq[Any]
+          val unpacked = evaluated.flatMap(sequence => {
+            WdlNamespace.load(wdl).workflows.head.inputs.get(name) match {
+              // TODO: Cromwell doesn't yet have compound types (e.g. Seq)
+              // TODO: so until it does just take the head
+              case Some(wdlType) => getSingleResult(sequence)
+              case None => getSingleResult(sequence)
+            }
+          })
+          (name, unpacked)
+        }
+      }
+    }
+  }
+
+  /**
+   * Iterate over the WDL inputs and ensure that (1) all necessary inputs are specified and (2) types are correct.
+   * @return map of from input name to error message (so if the map is empty, the config is valid)
+   */
+  def getWdlBindingErrors(configInputs: Map[String, Any], wdl: String) = {
+    WdlNamespace.load(wdl).workflows.head.inputs.map {
+      case (wdlName, wdlType) => {
+        val errorOption = configInputs.get(wdlName) match {
+          case Some(rawValue) => {
+            // TODO: currently Cromwell's exposed type-checking is not very robust.
+            // TODO: once Cromwell has a validation endpoint, use that; until then, don't bother checking types.
+            None
+          }
+          case None => Option("WDL binding failure: Input is missing from method config")
+        }
+        (wdlName, errorOption)
+      }
+    } collect { case (name, Some(error)) => (name, error) }
+  }
+
+  /**
+   * Get errors in both expression eval and WDL binding.
+   * @return map of from input name to error message (so if the map is empty, the config is valid)
+   */
+  def getValidationErrors(methodConfig: MethodConfiguration, entity: Entity, wdl: String, txn: RawlsTransaction) = {
+    val (resolved, unresolved) = resolveInputs(methodConfig, entity, wdl, txn).partition(_._2.isSuccess)
```

Comment:
I think I would prefer doing this in 1 loop instead of with partition. With the current code 1 problem will be a parse error will also be reported as a binding error. Another problem is that we are parsing the same wdl over and over again.

Let's talk through this a little more.